docs/developer/decision-records/2024-10-02-clustered-data-plane/README.md

```diff
@@ -0,0 +1,39 @@
+# Clustered data-plane
+
+## Decision
+
+We will make the data-plane being able to run in a clustered environment.
+
+## Rationale
+
+Currently, data-plane cannot run effectively in a clustered environment because:
+- there's no way to identify a specific replica that is running a data flow and "terminate/suspend" it
+- there's no way to re-start a data flow that was interrupted because the replica crashed
+
+## Approach
+
+We will provide this feature through the `DataPlaneStore` persistence layer.
+A `runtimeId` will be added in the `DataFlow`, and it will be set when it gets started with the replica's `runtimeId`.
+There will be a configured duration `T` (that can be in milliseconds, seconds at most. A good name for it needs to be found).
```

Comment:
how about flow lease? We may want to use the feature in the future to also check for stuck processes but we don't need that at the moment